interfaces/cython/cantera/_onedim.pyx

```diff
@@ -396,24 +399,54 @@ cdef class SymmetryPlane1D(Boundary1D):
 
 cdef class Surface1D(Boundary1D):
     """A solid surface."""
-    def __cinit__(self, *args, **kwargs):
-        self.surf = new CxxSurf1D()
+    def __cinit__(self, _SolutionBase phase, **kwargs):
+        self.surf = new CxxSurf1D(phase._base)
         self.boundary = <CxxBoundary1D*>(self.surf)
 
     def __dealloc__(self):
         del self.surf
 
 
 cdef class ReactingSurface1D(Boundary1D):
-    """A reacting solid surface."""
-    def __cinit__(self, *args, **kwargs):
-        self.surf = new CxxReactingSurf1D()
+    """A reacting solid surface.
+
+    :param phase:
+        The (surface) phase corresponding to the boundary
+
+    .. versionchanged:: 3.0
+
+        Starting in Cantera 3.0, parameter `phase` should reference surface instead of
+        gas phase.
+    """
+    def __cinit__(self, _SolutionBase phase, **kwargs):
+        if isinstance(phase, _SolutionBase) and phase.phase_of_matter != "gas":
+            self.surf = new CxxReactingSurf1D(phase._base)
+        else:
+            # legacy pathway - deprecation is handled in __init__
+            self.surf = new CxxReactingSurf1D()
         self.boundary = <CxxBoundary1D*>(self.surf)
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, _SolutionBase phase, name=None):
         self._weakref_proxy = _WeakrefProxy()
-        super().__init__(*args, **kwargs)
-        self.surface = None
+        if isinstance(phase, _SolutionBase):
+            if phase.phase_of_matter == "gas":
+                warnings.warn("Starting in Cantera 3.0, parameter 'phase' should "
+                    "reference surface instead of gas phase.", DeprecationWarning)
+                super().__init__(phase, name=name)
+            else:
+                sol = phase
+                gas = None
+                for val in sol._adjacent.values():
+                    if val.phase_of_matter == "gas":
+                        gas = val
+                        break
+                if gas is None:
+                    raise CanteraError("ReactingSurface1D needs an adjacent gas phase")
+                super().__init__(gas, name=name)
+        else:
+            super().__init__(phase, name=name)
```

Comment:
Is it actually possible to end up in this `else` clause?

Reply:
Nice catch - this was left over from an earlier version.